### PR TITLE
chore(link): add `meta` support

### LINKED ISSUE #2821: refactor: add support for `meta` in `@link`
Link doesn't support for additional meta information with the URLs. Let's add it, for eg:

```graphql
schema @link(
  src: "http://products:4000",
  type: SubGraph,
  meta: { 
    description: "Product service subgraph", 
    version: "1.2.0", 
    environment: "production"
  }
){
  query: Query
}
```

**Technical Requirements**
- Meta should accept any `JSON` scalar.
- Add concise tests for `meta`

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Issue Reference(s):**  
Fixes #2821

### CODE REVIEW

File: src/core/config/link.rs [hunk 1]
```diff
@@ -57,4 +57,7 @@ pub struct Link {
     /// The type of the link. It can be `Config`, or `Protobuf`.
     #[serde(default, skip_serializing_if = "is_default", rename = "type")]
     pub type_of: LinkType,
+    /// Metadata
```
Comment (tusharmath) [inline src/core/config/link.rs:60]:
```suggestion
    /// Additional metadata pertaining to the linked resource.
```

File: src/core/grpc/request_template.rs [hunk 2]
```diff
@@ -160,6 +160,7 @@ mod tests {
             id: Some(id.clone()),
             src: test_file.to_string(),
             type_of: LinkType::Protobuf,
+            meta: None,
```
Comment (tusharmath) [inline src/core/grpc/request_template.rs:163]: Not required here.

File: src/core/grpc/protobuf.rs [hunk 3]
```diff
@@ -266,6 +266,7 @@ pub mod tests {
             id: Some(id.clone()),
             src: path.to_string(),
             type_of: LinkType::Protobuf,
+            meta: None,
```
Comment (tusharmath) [inline src/core/grpc/protobuf.rs:269]: Not required here.

File: src/core/grpc/data_loader_request.rs [hunk 4]
```diff
@@ -73,6 +73,7 @@ mod tests {
             id: None,
             src: test_file.to_string(),
             type_of: LinkType::Protobuf,
+            meta: None,
```
Comment (tusharmath) [inline src/core/grpc/data_loader_request.rs:76]: Not required here.

File: src/core/grpc/request_template.rs (hunk 2 above)
Comment (ssddOnTop) [reply src/core/grpc/request_template.rs:163]: please elaborate

**Review by tusharmath — APPROVED** [review]

**Comment by tusharmath** [comment]
Update to docs is required.

**Pull request merged**